src/bootstrap/src/core/build_steps/compile.rs

```diff
@@ -2293,6 +2295,45 @@ impl Step for Assemble {
             }
         }
 
+        if builder.config.llvm_offload && !builder.config.dry_run() {
+            debug!("`llvm_offload` requested");
+            let offload_install = builder.ensure(llvm::OmpOffload { target: build_compiler.host });
+            if let Some(_llvm_config) = builder.llvm_config(builder.config.host_target) {
+                let src_dir = offload_install.join("lib");
+                let libdir = builder.sysroot_target_libdir(build_compiler, build_compiler.host);
+                let target_libdir =
+                    builder.sysroot_target_libdir(target_compiler, target_compiler.host);
+                let lib_ext = std::env::consts::DLL_EXTENSION;
+
+                let libenzyme = "libLLVMOffload";
```

Comment:
As noted in another comment, the logic for locating these components should ideally be localized within the offload step itself. It should then contain the paths to the dylibs (either just as a `Vec<PathBuf>`, or as three fields, depending on whether it is useful to distinguish them), and the code here in `Assembly` should just copy them to the correct directories.